scripts/linters/general_purpose_linter.py

```diff
@@ -539,26 +543,35 @@ def check_bad_pattern_in_file(filepath, file_content, pattern):
             Object containing details for the pattern to be checked.
 
     Returns:
-        bool. True if there is bad pattern else false.
+        failed: bool. True if there is bad pattern else false.
+        summary_messages: list(str). Summary of failed messages.
     """
+    summary_messages = []
+    failed = False
     regexp = pattern['regexp']
     if not (any(filepath.startswith(excluded_dir)
                 for excluded_dir in pattern['excluded_dirs'])
             or any(filepath.endswith(excluded_file)
                    for excluded_file in pattern['excluded_files'])):
         bad_pattern_count = 0
-        for line_num, line in enumerate(file_content):
-            stripped_line = line.rstrip()
+        for line_num, line in enumerate(file_content, 1):
+            if line.endswith('\n'):
+                stripped_line = line[:-1]
```

Comment:
Why this changes?

Reply:
This is bug introduced by me in a past PR, which is fixed here.

Reply:
Why you are fixing it here, We should have created a separate PR to fix it and  get it merged quickly! (It's creating a problem on develop isn't it)?

Reply:
No, it is not creating problems on develop, it will only cause problem if we have TODO as the last of the comment. I fixed it here becuase I found this while writing test for TODO check.

Reply:
I see thanks for the explanation.